source/menus/dungeon_menu.cpp

```diff
@@ -0,0 +1,195 @@
+    {"boss key", BOSS_KEY_INDEX, "get the boss key",true, &l_bosskeyFlag},
+    {"map", MAP_INDEX, "get the map", true, &l_mapFlag},
+    {"compass", COMPASS_INDEX, "get the compass",true, &l_compassFlag},
+    {"partner", PARTNER_INDEX, "Medli/Makar in ET/WT",true, &l_partner},
+};
+
+void dungeonItemSwitch(int stage, int flag) {
+    if (dComIfGs_isDungeonItem(stage, flag)) {
+        dComIfGs_offDungeonItem(stage, flag);
+    }
+    else {
+        dComIfGs_onDungeonItem(stage, flag);
+    }
+}
+
+const char* get_dun_name(u8 area_id) {
+    switch (area_id) {
+        case 3:
+            return "DRC";
+            break;
+        case 4:
+            return "FW";
+            break;
+        case 5:
+            return "TOTG";
+            break;
+        case 6:
+            return "ET";
+            break;
+        case 7:
+            return "WT";
+            break;
+        default:
+            return "DRC";
+    }
+}
+
+u8 selectDungeon() {
+    u8 stage = 3;
+    Cursor::moveListSimple(stage);
+    if (stage == 2) {
+        stage = 7;
+    }
+    if (stage > 7) {
+        stage = 3;
+    }
+    return stage;
+}
+
+void modifyKeys(int stage_id) {
+    u8 amount = dComIfGs_getKeyNum(stage_id);
+    Cursor::moveListSimple(amount);
+    if (amount == 0xFF) {
+        amount = 99;
+    }
+    if (amount > 99) {
+        amount = 0;
+    }
+    dComIfGs_setKeyNum(stage_id, amount);
+}
+
+
+void DungeonMenu::draw() {
+    cursor.setMode(Cursor::MODE_LIST);
+
+    if (GZ_getButtonTrig(GZPad::B)) {
+        GZ_setMenu(GZ_FLAGS_MENU);
+        return;
+    }
+
+    u8 area_id = 3;
+    switch (l_selDun) {
+    case 0:
+        area_id = 3;
+        break;
+    case 1:
+        area_id = 4;
+        break;
+    case 2:
+        area_id = 5;
+        break;
+    case 3:
+        area_id = 6;
+        break;
+    case 4:
+        area_id = 7;
+        break;
+    }
+
+    l_mapFlag = dComIfGs_isDungeonItem(area_id, 0);
+    l_compassFlag = dComIfGs_isDungeonItem(area_id, 1);
+    l_bosskeyFlag = dComIfGs_isDungeonItem(area_id, 2);
+    l_medli = (dComIfGs_isEventBit(0x1620) && dComIfGs_isEventBit(0x1608) &&
+               dComIfGs_isEventBit(0x2920) && dComIfGs_isEventBit(0x2E04));
+    l_makar = (dComIfGs_isEventBit(0x1610) && dComIfGs_isEventBit(0x1604) && dComIfGs_isEventBit(0x2910));
+    if (area_id == 6) { l_partner = l_medli; }
+    if (area_id == 7) { l_partner = l_makar; }
+    int MAX_MOVE;
+    if ((area_id == 6) || (area_id == 7)) {
+        MAX_MOVE = LINE_NUM;
+    } else {
+        MAX_MOVE = LINE_NUM - 1;
+    }
+    switch (cursor.y) {
+    case PICK_DUNGEON_INDEX:
+        cursor.x = l_selDun;
+        cursor.move(5, MAX_MOVE);
+
+        if (cursor.y == PICK_DUNGEON_INDEX) {
+            l_selDun = cursor.x;
+        }
+        l_keyNum = dComIfGs_getKeyNum(area_id);
+        selectDungeon();
+        break;
+    case MODIFY_KEYS_INDEX:
+        cursor.move(0, LINE_NUM);
```

Comment:
Should this say "MAX_MOVE" instead of "LINE_NUM"?

Reply:
It doesn't seem to matter either way, but I can change it for consistency